simulation/sub8_gazebo/nodes/fake_perception.py

```diff
@@ -0,0 +1,125 @@
+
+import sys
+import time
+from tf import transformations
+import yaml
+import xml.etree.ElementTree
+rospack = rospkg.RosPack()
+#Find a way to parse through the yaml correctly. Might be unneccessary but nice.
+config_file = os.path.join(rospack.get_path('sub8_missions'), 'sub8', 'vision_proxies.yaml')
+f = yaml.load(open(config_file, 'r'))
+launch_file = os.path.join(rospack.get_path('sub8_gazebo'), 'launch', 'duck.launch')
+launch = xml.etree.ElementTree.parse(launch_file).getroot()
+
+fprint = text_effects.FprintFactory(title="SIMULATOR").fprint
+
+def handle_fake_perception(extra,resp):
+    now = rospy.get_rostime()
+    k = np.uint32(0)
+    if extra != '':
+        resp = extra
+    if resp == '':
+        print "NO TARGET"
+        sys.exit(0)
+    for node in launch.iter('node'):
+        name = node.get('name')
+        if name == resp:
+            v = node.get('args')
+            point = getCoords(resp,v)
+            q = getOrient(resp,v)
+
+    pose_stamp = PoseStamped(header = Header(seq = k, stamp = now, frame_id = "/map"),
+        pose = Pose(position=point,
+        orientation=Quaternion(*q)))
+    covariance_diagonal = Vector3(0,0,0)
+    found = True
+    resp2 = VisionRequestResponse(pose_stamp,covariance_diagonal,found)
+    return resp2
+def getOrient(name,v):
+    #Find out X orient
+    j = v.find('-X')
+    start = j+3
+    end = v.find(' ', start)
+    XX = (float)(v[start:end])
+
+    #Find our Y orient
+    j = v.find('-Y')
+    start = j+3
+    end = v.find(' ', start)
+    YY = (float)(v[start:end])
+
+    #Find our Z orient
+    j = v.find('-Z')
+    start = j+3
+    end = v.find(' ', start)
+    ZZ = (float)(v[start:end])
+
+    response = transformations.quaternion_from_euler(XX,ZZ,YY)
+    return response
+def getCoords(name,v):
+    #Im going to need to edit our launch file to match all the names of the objects in the launch file to the names of the target objects in this
+    #On top of that I need to centralize all pose and orientation numbers into the launch file, moving them out of the STL files.
+    #After that this should be relatively easy to implement, but time consuming.
+    #IF target_name != ''
+        #run get coords while target_name != to previous target_name
+    #Find our x.
+    j = v.find('-x')
+    start = j+3
+    end = v.find(' ', start)
+    x = (float)(v[start:end])
+
+    #Find our y
+    j = v.find('-y')
+    start = j+3
+    end = v.find(' ', start)
+    y = (float)(v[start:end])
+
+    #Find our z
+    j = v.find('-z')
+    start = j+3
+    end = v.find(' ', start)
+    z = (float)(v[start:end])
+    return Point((x-13.0),(y-24.0),(-1))
+def set_geometry(req):
+    return {'success': True}
+def vision_cb_2D():
+    return True
+def start(resp):
+    return SetBoolResponse(True,"")
+def init_service(name,target):
+    #Generates the important service
+    serv = rospy.Service('/vision/'+name+'/pose',VisionRequest, lambda h: handle_fake_perception(h.target_name,target))
```

Comment:
The CI will catch this, but if a variable is never used you don't need to declare it.
```
rospy.Service('/vision/'+name+'/pose',VisionRequest, lambda h: handle_fake_perception(h.target_name,target))
rospy.Service('/vision/'+name+'/set_geometry', SetGeometry, set_geometry)
...
```
Would work fine here